src/main/cpp/include/query_operations/variant_operations.h

```diff
@@ -88,6 +88,76 @@ class RemappedVariant : public RemappedDataWrapperBase
     unsigned m_queried_field_idx;
 };
 
+typedef std::tuple<const std::vector<int>*, uint64_t, bool> GenotypeForMinValueResultTuple;
+
+template<class DataType>
+class GenotypeForMinValueTracker : public RemappedDataWrapperBase
+{
+  public:
+    GenotypeForMinValueTracker()
+    {
+      reset();
+    }
+    void reset()
+    {
+      m_current_min_value = std::numeric_limits<DataType>::max();
+      m_found_one_valid_value = false;
+    }
+    void* put_address(uint64_t input_call_idx, unsigned allele_or_gt_idx)
+    {
+      throw VariantOperationException("put_address() undefined for class GenotypeForMinValueTracker and shouldn't be called");
+      return 0;
+    }
+    void track_minimum(const std::vector<DataType>& input_data,
+        std::vector<int>& allele_idx_vec_for_current_gt_combination);
+    /*
+     * Wrapper around GenotypeForMinValueTracker - just for compatibility with remap_data_based_on_genotype_general
+     * remap_data_based_on_genotype_general should be fixed in general
+     */
+    static void determine_allele_combination_and_genotype_index_for_min_value(const std::vector<DataType>& input_data,
```

Comment:
Too many args for a function, do you think you can create structures with related fields?

Reply:
Yep - the whole set of functions related to remap\_ suck in terms of interface/arguments :(

I have to restructure that whole code segment to make it cleaner and usable. For now, this will break other components